tinygrad/runtime/ops_metal.py

```diff
@@ -60,7 +60,13 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int]=(1,1,1), local_size:Tup
 class MetalAllocator(LRUAllocator):
   def __init__(self, device:MetalDevice):
     self.device:MetalDevice = device
+    self.track_cross_device: Set[MetalDevice] = set()
```

Comment:
Huh, how do devices end up in this set?

Reply:
<img width="978" alt="image" src="https://github.com/tinygrad/tinygrad/assets/83587632/6154b404-bb12-445b-949c-3977a0b0d83e">
in device.py